src/controllers/resource.ts

```diff
@@ -37,14 +43,41 @@ export class ResourceController {
 				return true;
 			})
 			.withMessage('name, type and content are required'),
+		check('content').exists().isString().withMessage(Messages.Invalid),
 		check('name').optional().isString().withMessage(Messages.Invalid),
 		check('type').optional().isString().withMessage(Messages.Invalid),
-		check('content').exists().isString().withMessage(Messages.Invalid),
+		check('version').optional().isString().withMessage(Messages.Invalid),
 		check('relativeDidUrl').optional().isString().contains('/resources/').withMessage(Messages.InvalidDidUrl),
 		check('alsoKnownAs').optional().isArray().withMessage(Messages.Invalid),
 		check('alsoKnownAs.*.uri').isString().withMessage(Messages.Invalid),
 		check('alsoKnownAs.*.description').isString().withMessage(Messages.Invalid),
 	];
+	public static updateResourceValidator = [
+		check('did').exists().isString().contains('did:cheqd').withMessage(Messages.InvalidDid),
+		check('jobId')
+			.custom((value, { req }) => {
+				if (!value && !(req.body.name && req.body.type && req.body.content)) return false;
+				return true;
+			})
+			.withMessage('name, type and content are required'),
+		check('name').optional().isString().withMessage(Messages.Invalid),
+		check('type').optional().isString().withMessage(Messages.Invalid),
+		check('content')
+			.exists()
+			.isArray()
+			.custom((value) => {
+				if (value.length !== 1) return false;
+				if (typeof value[0] !== 'string') return false;
+				return true;
+			})
+			.withMessage('The content array must be provided and must have exactly one string'),
+		check('relativeDidUrl').exists().isString().contains('/resources/').withMessage(Messages.InvalidDidUrl),
```

Comment:
Can we make this optional relativeDidUrl 

if not provided we update the latest resource, by fetching the resource id using the name and type

Reply:
Get using ?resourceName=Score%20schema-default&resourceType=anonCredsCredDef&resourceMetadata=true
Validate using the nextVersionId=null